src/rust/engine/process_execution/src/local.rs

```diff
@@ -878,9 +878,8 @@ pub fn setup_run_sh_script(
   let full_script = format!(
     "#!/bin/bash
 # This command line should execute the same process as pants did internally.
-export {stringified_env_vars}
 cd {stringified_cwd}
-{stringified_command_line}
+env -i {stringified_env_vars} {stringified_command_line}
```

Comment:
I've done this by hand myself reproing user issues and it worked fine. Did you double check this still works?

Reply:
Yep, ran a few different command (test, lint, fmt) with `--keep-sandboxes=always`, cd'd into the sandbox and ran `__run.sh` to ensure that the process still does the thing we expect it to do.

Reply:
I just checked `env -i bash -c 'echo $HOME.'` -> ".", but I did not gut check a sandbox `__run.sh` using this change.